cpp/include/cudf/detail/copy.hpp

```diff
@@ -88,6 +88,30 @@ std::unique_ptr<column> shift(
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Performs segmented shifts for specified values.
+ *
+ * For each segment, `i`th element is determined by the `i - offset`th element
+ * of the segment. If `i - offset < 0 or >= segment_size`, the value is determined by
+ * @p fill_value .
+ *
+ * @param segmented_values Segmented column, specifed by @p segment_offsets
+ * @param segment_offsets Each segment's offset of @p segmented_values . A list of offsets
+ * with size `num_segments + 1`. The size of each segment is `segment_offsets[i+1] -
+ * segment_offsets[i]`.
+ * @param offset The offset by which to shift the input
+ * @param fill_value Fill value for indeterminable outputs
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr Device memory resource used to allocate the returned table and columns' device memory
+ */
```

Comment:
Would be nice if you can provide a simple example in the doxygen.

Reply:
Added.